src/Message/Formatter.php

```diff
@@ -235,15 +235,16 @@ private function getTrace(Message $message): string
     {
         /** @psalm-var Backtrace $traces */
         $traces = $message->context('trace', []);
-
-        foreach ($traces as $key => $trace) {
-            if (isset($trace['file'], $trace['line'])) {
-                $traces[$key] = "in {$trace['file']}:{$trace['line']}";
-            }
+        if (empty($traces)) {
+            return '';
         }
 
-        /** @var string[] $traces */
-        return empty($traces) ? '' : "trace:\n    " . implode("\n    ", $traces);
+        $lines = array_map(
+            static fn (array $trace): string => "in {$trace['file']}:{$trace['line']}",
```

Comment:
`file` and `line` keys are always present, right?

Reply:
By [this type](https://github.com/yiisoft/log/blob/b96afb70db157ed7d79593f9a8ad49ac95a366ea/src/Message.php#L17)— yes ([debug_backtrace()](https://www.php.net/manual/ru/function.debug-backtrace.php) is used by default)

But user can pass own incorrect trace. We don't check this anywhere.